### PR TITLE
Show `Connection Issues` notice on live activity when it's stale

### PR DESCRIPTION
**Pull request by guytepper**
IMAGE 2023-06-16 14:25:07

### CODE REVIEW

File: ios/BetterRailWidget/Live Activity/BetterRailLiveActivity.swift [hunk 1]
```diff
@@ -44,12 +42,19 @@ struct BetterRailLiveActivity: Widget {
...
+              .foregroundColor(tintColor(context: context))
+          } else {
+            Text("\(String(getMinutesLeft(targetDate: endDate))) min")
+              .foregroundColor(tintColor(context: context))
+              .accessibilityLabel(vm.status == .waitForTrain ? "time left depart" : "time left arrival")
+              .contentTransition(.numericText())
+          }
         } minimal: {
-          CircularProgressView(vm: vm, content: .time, tintColor: tintColor(context: context))
+          CircularProgressView(vm: vm, content: vm.isStale ? .icon : .time, tintColor: tintColor(context: context))
```
Comment (planecore) [inline ios/BetterRailWidget/Live Activity/BetterRailLiveActivity.swift:57]: I prefer the stale icon logic to be in `CircularProgressView` with the rest of the conditions

File: ios/BetterRailWidget/Live Activity/Views/Lock Screen/InTransitView.swift [hunk 2]
```diff
@@ -14,22 +14,24 @@ struct LockScreenInTransitView: View {
...
             Text("arrived at").font(.subheadline2)
+          } else if (vm.isStale) {
+            Text("headed to")
+              .fontWeight(vm.isRTL ? .regular : .light)
+              .font(vm.isRTL ? .subheadline2 : .caption)
           } else {
             Text("next station")
               .fontWeight(vm.isRTL ? .regular : .light)
               .font(vm.isRTL ? .subheadline2 : .caption)
           }
```
Comment (planecore) [inline ios/BetterRailWidget/Live Activity/Views/Lock Screen/InTransitView.swift:25]:
```suggestion
          } else {
            Text(vm.isStale ? "headed to" : "next station")
              .fontWeight(vm.isRTL ? .regular : .light)
              .font(vm.isRTL ? .subheadline2 : .caption)
          }
```

File: ios/BetterRailWidget/Live Activity/Views/TimeInformation.swift [hunk 3]
```diff
@@ -22,13 +22,38 @@ struct TimeInformation: View {
...
         VStack(alignment: .trailing) {
           Text("ARRIVAL TIME").font(.caption)
           Text(formatDateHour(targetDate)).font(.system(size: 18, weight: .bold, design: .rounded))
         }
-        
+      }
+      else if (vm.isStale) {
+        staleInformation
       }
       else {
```
Comment (planecore) [inline ios/BetterRailWidget/Live Activity/Views/TimeInformation.swift:58]:
```suggestion
      } else {
```

**Review by planecore — CHANGES_REQUESTED** [review]

File: ios/BetterRailWidget/Live Activity/Views/TimeInformation.swift [hunk 4]
```diff
@@ -22,13 +22,38 @@ struct TimeInformation: View {
   
   var placement: ViewPlacement = .lockScreen
   
+  var staleInformation: some View {
+    VStack(alignment: .trailing) {
+      HStack(alignment: .firstTextBaseline, spacing: 4) {
+        if (placement == .lockScreen) {
+          Text("arrive")
```
Comment (planecore) [inline ios/BetterRailWidget/Live Activity/Views/TimeInformation.swift:29]: We should show "depart" text when the user is waiting for train or in exchange.

File: ios/BetterRailWidget/Live Activity/Views/ActivityHeader.swift [hunk 5]
```diff
@@ -4,14 +4,29 @@ struct ActivityHeader: View {
...
+        
+      if (vm.isStale) {
+        HStack(spacing: 4.0) {
+          Image(systemName: "exclamationmark.circle.fill")
+            .font(.caption2)
+          
+          Text("connection issues")
+            .font(.caption2)
+        }
+        .foregroundColor(.orange).fontWeight(.heavy)
```
Comment (planecore) [inline ios/BetterRailWidget/Live Activity/Views/ActivityHeader.swift:28]: Changing to red might be more appropriate:

IMG_6179

Comment (guytepper) [reply ios/BetterRailWidget/Live Activity/Views/ActivityHeader.swift:28]: it'll be mixed up with the delay color :/

File: ios/BetterRailWidget/Live Activity/BetterRailLiveActivity.swift (hunk 1 above)
Comment (guytepper) [reply ios/BetterRailWidget/Live Activity/BetterRailLiveActivity.swift:57]: why? imo it's better to have this controlled by the consumer view

File: ios/BetterRailWidget/Live Activity/Views/TimeInformation.swift (hunk 3 above)
Comment (guytepper) [reply ios/BetterRailWidget/Live Activity/Views/TimeInformation.swift:58]: ?

File: ios/BetterRailWidget/Live Activity/BetterRailLiveActivity.swift (hunk 1 above)
Comment (planecore) [reply ios/BetterRailWidget/Live Activity/BetterRailLiveActivity.swift:57]: There are many conditions where the time fallbacks to icon. This is one of them. Look in the `CircularProgressView` file.

File: ios/BetterRailWidget/Live Activity/Views/TimeInformation.swift (hunk 3 above)
Comment (planecore) [reply ios/BetterRailWidget/Live Activity/Views/TimeInformation.swift:58]: Don't start a new line after the closing parenthesis. Close and open in the same line.

**Review by planecore — APPROVED** [review]

**Pull request merged**